Framework/API/test/MatrixWorkspaceTest.h

```diff
@@ -1486,8 +1514,51 @@ class MatrixWorkspaceTestPerformance : public CxxTest::TestSuite {
     TS_ASSERT_DELTA(result, 5214709.740869, 1e-6);
   }
 
+  /*
+   * Rotate a bank in the workspace and read the positions out again. Very typical.
+   */
+  void test_rotate_bank_and_read_positions() {
+
+    using namespace Mantid::Geometry;
+    using namespace Mantid::Kernel;
+
+    // Rotate the bank
+    ComponentHelper::rotateComponent(
+        *m_sansBank, *m_paramMap, m_zRotation,
+        Mantid::Geometry::ComponentHelper::Relative);
+
+    V3D pos;
+    for (size_t i = 1; i < m_workspaceSans.getNumberHistograms(); ++i) {
+
+      pos += m_workspaceSans.getDetector(i)->getPos();
+    }
+  }
+
+  /*
+   * Move a bank in the workspace and read the positions out again. Very typical.
+   */
+  void test_move_bank_and_read_positions() {
+
+    using namespace Mantid::Geometry;
+    using namespace Mantid::Kernel;
+
+    // move the bank
+    ComponentHelper::moveComponent(*m_sansBank, *m_paramMap, m_pos,
+                                   Mantid::Geometry::ComponentHelper::Relative);
+
+    V3D pos;
+    for (size_t i = 1; i < m_workspaceSans.getNumberHistograms(); ++i) {
+      pos += m_workspaceSans.getDetector(i)->getPos();
+    }
+  }
+
```

Comment:
Great! I can later simply copy these tests and use `SpectrumInfo`, which should give a read speed-up, i.e., we then have three cases to compare.